crates/pessimistic-proof/src/imported_bridge_exit.rs

```diff
@@ -1,113 +1,221 @@
-/// the current network.
 #[derive(Debug, Clone, Serialize, Deserialize)]
-pub struct ImportedBridgeExit {
-    /// The bridge exit initiated on another network, called the "sending"
-    /// network. Need to verify that the destination network matches the
-    /// current network, and that the bridge exit is included in an imported
-    /// LER
-    pub bridge_exit: BridgeExit,
-    /// The Imported Local Exit Root for the Local Exit Tree containing this
-    /// bridge exit.
-    pub imported_local_exit_root: Digest,
-    /// The inclusion proof of the imported bridge exit in the sending local
-    /// exit root.
-    pub inclusion_proof: LETMerkleProof<Keccak256Hasher>,
-    /// The inclusion proof of the LER to the Rollup Exit Root and the Rollup
-    /// Exit Root.
-    pub inclusion_proof_rer: Option<(LETMerkleProof<Keccak256Hasher>, Digest)>,
-    /// The global index of the imported bridge exit.
-    pub global_index: GlobalIndex,
+pub struct L1InfoTreeLeafInner {
+    pub ger: Digest,
+    pub block_hash: Digest,
+    pub timestamp: u64,
 }
 
-impl ImportedBridgeExit {
-    /// Creates a new [`ImportedBridgeExit`].
-    pub fn new(
-        bridge_exit: BridgeExit,
-        imported_local_exit_root: KeccakDigest,
+impl L1InfoTreeLeafInner {
+    fn hash(&self) -> Digest {
+        keccak256_combine([
+            self.ger.as_slice(),
+            self.block_hash.as_slice(),
+            &self.timestamp.to_be_bytes(),
+        ])
+    }
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct L1InfoTreeLeaf {
+    pub l1_info_tree_index: u32,
+    pub rer: Digest,
+    pub mer: Digest,
+    pub inner: L1InfoTreeLeafInner,
+}
+
+impl L1InfoTreeLeaf {
+    pub fn hash(&self) -> Digest {
+        self.inner.hash()
+    }
+}
+
+#[derive(Debug, Error)]
+pub enum Error {
+    #[error("Mismatch between the global index and the inclusion proof.")]
+    MismatchGlobalIndexInclusionProof,
+    #[error("Mismatch between the provided L1 root and the inclusion proof.")]
+    MismatchL1Root,
+    #[error("Mismatch on the MER between the L1 leaf and the inclusion proof.")]
+    MismatchMER,
+    #[error("Mismatch on the RER between the L1 leaf and the inclusion proof.")]
+    MismatchRER,
+    #[error("Invalid merkle path from the leaf to the LER.")]
+    InvalidMerklePathLeafToLER,
+    #[error("Invalid merkle path from the LER to the RER.")]
+    InvalidMerklePathLERToRER,
+    #[error("Invalid merkle path from the GER to the L1 Info Root.")]
+    InvalidMerklePathGERToL1Root,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct MerkleProof {
+    pub proof: LETMerkleProof<Keccak256Hasher>,
+    pub root: Digest,
+}
+
+impl MerkleProof {
+    pub fn verify(&self, leaf: Digest, leaf_index: u32) -> bool {
+        self.proof.verify(leaf, leaf_index, self.root)
+    }
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub enum Claim {
+    Mainnet(Box<ClaimFromMainnet>),
```

Comment:
`Mainnet` isn't the right terminology I guess. If we deploy on testnet or devnet, does it mean that it target mainnet?
Shouldn't it be L1? or Layer1 ?

Reply:
Since this is for the inclusion proofs, I tried to map that to the terminology used within the Global Exit Tree data structure:

- `ClaimFromRollup`: inclusion proofs within the `RollupExitTree`
- `ClaimFromMainnet`: inclusion proof within the `MainnetExitTree`

The `MainnetExitTree` remains called like that even if we deal with testnet/devnet data as part of the lxly terminology. Maybe we could change that as part of v0.3.0

Reply:
Mmh I see.
For me `mainnet`, `testnet` and `devnet` are environment ^^